tests/unit/test_main.py

```diff
@@ -0,0 +1,151 @@
+"""Test Main methods."""
+from unittest import TestCase
+from unittest.mock import patch
+from napps.kytos.of_core.v0x04.flow import Flow as Flow04
+from napps.amlight.noviflow.of_core.v0x04.action import (
+    NoviActionPushInt,
+    NoviActionSetBfdData,
+    NoviActionAddIntMetadata,
+    NoviActionPopInt,
+    NoviActionSendReport,
+)
+
+from kytos.lib.helpers import (
+    get_controller_mock,
+    get_switch_mock,
+)
+
+
+class TestMain(TestCase):
+    """Tests for the Main class."""
+
+    def setUp(self):
+        patch("kytos.core.helpers.run_on_thread", lambda x: x).start()
+        # pylint: disable=import-outside-toplevel
+        from napps.amlight.noviflow.main import Main
+
+        self.addCleanup(patch.stopall)
+
+        controller = get_controller_mock()
+        self.napp = Main(controller)
+        self.napp.register_actions()
+        self.mock_switch = get_switch_mock("00:00:00:00:00:00:00:01", 0x04)
+
+    def test_create_noviactions(self):
```

Comment:
I'm planning to increase test cov to also add tests for unpacking on this issue here https://github.com/kytos-ng/noviflow/issues/4